internal/app/components/courses/courseRecords.go

```diff
@@ -66,3 +68,15 @@ func (cr CourseRecords) Exclude(records CourseRecords) CourseRecords {
 	}
 	return result
 }
+
+func (cr CourseRecord) IsPrereqSatisfied(pastRecords *CourseRecords) bool {
```

Comment:
i feel like you can merge the two if statements in the function, or at least reuse prereqs var in the return statement

Reply:
true